views/chain.py

```diff
@@ -0,0 +1,37 @@
+import time
+
+from flask import request
+
+from flask_app import app
+from models import IndexerState, Session
+
+
+@app.route("/wait-for-block")
+def wait_for_block():
+    """Wait for a block to get processed before returning.
+
+    Will be called as `GET /wait-for-block?block=123`,
+    where 123 is the block we need to wait for.
+    """
+    with Session() as s:
+        block = request.args.get("block")
+
+        if not block:
+            return {"ok": False, "error": "Block param is mandatory."}
+
+        block = int(block)
```

Comment:
what if parsing fails?

Reply:
It will throw an error that will get translated into an HTTP 500 error.

Reply:
Could be nice to throw a JSON error as well, up yo you.

Reply:
Added JSON error response message and also different error codes (for when we will add frontend retries based on error codes).